server/client.go

```diff
@@ -3587,22 +3624,35 @@ func (c *client) processServiceImport(si *serviceImport, acc *Account, msg []byt
 	} else if si.usePub {
 		to = string(c.pa.subject)
 	}
-
 	// Now check to see if this account has mappings that could affect the service import.
 	// Can't use non-locked trick like in processInboundClientMsg, so just call into selectMappedSubject
 	// so we only lock once.
 	to, _ = si.acc.selectMappedSubject(to)
 
-	// Save off some of our state.
-	oreply, oacc, opsi := c.pa.reply, c.acc, c.pa.psi
-	c.pa.reply = nrr
-	c.pa.psi = si
+	// Copy our pubArg and account
+	pacopy := c.pa
```

Comment:
But that is not a deep copy. Meaning that `int` etc.. will be copied, but reference to slices will be the same. So if c.pa.<any slice> change, then pacopy's slices will reflect the change. Is that ok?

Reply:
I don't change the strings or byte slices, but can replace them in the copy of c.pa which will be put back when done.